DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,178 @@
+export INTEGRATION=true
+
+# Set up docker ( this will build, install into Logstash )
+scripts/opensearch/docker-setup.sh
+
+# Run tests
+scripts/opensearch/docker-run.sh
+```
+
+To perform unit tests, set `INTEGRATION=false` and re-run the `docker-run.sh`.
+You don't have to set up docker again.
+
+2. Tests against OpenDistro clusters.
+
+```bash
+# Set up Environment variable for docker to pull your test environment
+export LOGSTASH_VERSION=7.13.2 # mandatory 
+export OPENDISTRO_VERSION=1.13.2 # will use latest if not specified.
+export INTEGRATION=false
+
+# Set up docker ( this will build, install into Logstash )
+scripts/opendistro/docker-setup.sh
+
+# Run tests
+scripts/opendistro/docker-run.sh
+```
+To perform unit tests, set `INTEGRATION=false` and re-run the `docker-run.sh`.
+You don't have to set up docker again.
+
+### Run plugin in Logstash
+
+#### 2.1 Run in a local Logstash clone
+
+1. Edit Logstash `Gemfile` and add the local plugin path, for example:
+
+   ```ruby
+   gem "logstash-output-opensearch", :path => "/your/local/logstash-output-opensearch"
+   ```
+
+2. Install the plugin:
+
+   ```sh
+   # Logstash 2.3 and higher
+   bin/logstash-plugin install --no-verify
+
+   # Prior to Logstash 2.3
+   bin/plugin install --no-verify
+   ```
+
+3. Run Logstash with your plugin:
+
+   ```sh
+   bin/logstash -e 'output {opensearch {}}'
+   ```
+
+At this point any modifications to the plugin code will be applied to this local Logstash setup. After modifying the plugin, simply re-run Logstash.
+
+#### 2.2 Run in an installed Logstash
+
+Build the gem locally and install it using:
+
+1. Build your plugin gem:
+
+   ```sh
+    gem build logstash-output-logstash.gemspec
+   ```
+   That’s it! Your gem should be built and be in the same path with the name
+    ```
+    Successfully built RubyGem
+    Name: logstash-output-opensearch
+    Version: 1.0.0
+    File: logstash-output-opensearch-1.0.0.gem
+    ```
+   The s.version number from your gemspec file will provide the gem version, in this case, 1.0.0.
+
+
+2. Install the plugin from the Logstash home:
+
+   ```sh
+   # Logstash 2.3 and higher
+   bin/logstash-plugin install ~/workspace/logstash-output-opensearch/logstash-output-opensearch-1.0.0.gem
+
+   # Prior to Logstash 2.3
+   bin/plugin install ~/workspace/logstash-output-opensearch/logstash-output-opensearch-1.0.0.gem
+   ```
+   
+   After running this, you should see the following feedback from Logstash to test the installation
+    ```bash
+    validating ~/workspace/logstash-output-opensearch/logstash-output-opensearch-1.0.0.gem >= 0
+    Valid logstash plugin. Continuing...
+    Successfully installed 'logstash-output-opensearch' with version '1.0.0' 
+   ```
+
+3. Start Logstash and test the plugin.
+
+    ```bash
+   bin/logstash-plugin list 
+   ```
+
+## Configuration for Logstash Output OpenSearch Plugin
+
+To run the Logstash Output Opensearch plugin, simply add a configuration following the below documentation.
```

Comment:
Simply is not a great word for this type of document.

Reply:
Ack